src/iwant_bot/start.py

```diff
@@ -24,64 +51,336 @@ def format_message(message):
+          # body=format_response(
+          #   "text": f"{', '.join(body['activity'])}! I am looking for someone for {time_minutes} minutes."
+          #    "attachments":json.dumps(iwant_duration_buttons(body))
+          # )
+       # )
 
-async def handle_post(request):
-    body = await request.post()
-    print(request.method)
-    if check_post_request(body):
-        user_id = body['user_id']
-        user_name = body['user_name']
-        text = body['text']
-        try:
-            trigger_word = body['trigger_word']
-        except KeyError:
-            print('No trigger_word was used.')
-            trigger_word = None
     else:
-        return web.Response(text='No valid Slack POST request.\n')
+        return web.json_response(body=format_response(iwant_help_message()))
+
+
+def iwant_duration_buttons(body):
+    """Create message with buttons to adjust request duration"""
+    time_minutes = round((body['duration_ts'] - body['incoming_ts']) / 60)
+    text = f"{', '.join(body['activity'])}! I am looking for someone for {time_minutes} minutes."
+    attachment = [
+        {
+            'text': 'Would you like to change the duration to:',
+            'callback_id': f"{body['callback_id']}",
+            'fallback': 'Never mind.',
+            #'color' : '#3AA3E3',
+            'attachment_type': 'default',
+            'actions': [
+                {
+                    'name': 'Time',
+                    'text': '30 min',
+                    'type': 'button',
+                    'value': '1800'
+                },
+                {
+                    'name': 'Time',
+                    'text': '1 hour',
+                    'type': 'button',
+                    'value': '3600'
+                },
+                {
+                    'name': 'Time',
+                    'text': 'Cancel',
+                    'type': 'button',
+                    'value': '0'
+                }
+            ]
+        }
+    ]
+    return {'text': text, 'attachments': attachment}
+
+
 
-    if trigger_word == 'repeat':
-        message = format_response(f'{user_name} wants me to {text}')
-    elif trigger_word == 'whoami':
-        message = format_response(f'You are {user_name} with id {user_id}')
+def iwant_activity(body: dict) -> bool:
+    """Search the text for activities.
+    Suppose a lot of assumptions...
+    ...nobody with the name @coffee, activity 'invite' etc."""
+    any_activity = False
+    body['activity'] = []
+    body['callback_id'] = []
+    for activity in _iwant_activities:
+        if re.search(activity, body['text'].lower()):
+            any_activity = True
+            body['duration_ts'] = body['incoming_ts'] + iwant_duration(body['text'])
+            body['activity'].append(activity)
+            body['callback_id'].append(create_iwant_request(body))
+            iwant_invite(body)
+    return any_activity
+
+
+def iwant_duration(text: str) -> float:
```

Comment:
Name `parse_duration_from_request_message_string` would be more descriptive for the function.